nautobot/dcim/forms.py

```diff
@@ -1350,6 +1358,34 @@ def clean_device_type(self):
 
         return data
 
+    def clean_type(self):
+        data = self.cleaned_data["type"]
+
+        try:
+            choices = self.Meta.model._meta.get_field("type").choices
+        except FieldDoesNotExist:
+            # Do we have a `DeviceComponent` subclass that doesn't define a `type` field?
```

Comment:
`InventoryItem`

Reply:
Actually this should be referring to a `DeviceComponentTemplate` (we don't yet have an `InventoryItemTemplate` class) but this is also true for `DeviceBayTemplate` not having a `type`. I'll update the comment.

Reply:
Actually I'll remove the try/except here as `clean_type()` should only be called automatically if there is a `type` field on the form.